code/eyetracking/eyetrackingrun.py

```diff
@@ -0,0 +1,781 @@
+                "gy_right": "eye1_y_coordinate",
+                "pa_right": "eye1_pupil_size",
+                "px_right": "eye1_pupil_x_coordinate",
+                "py_right": "eye1_pupil_y_coordinate",
+                "hx_right": "eye1_head_x_coordinate",
+                "hy_right": "eye1_head_y_coordinate",
+                "rx": "screen_pixel_per_degree_x",
+                "ry": "screen_pixel_per_degree_y",
+                "gxvel_right": "eye1_x_velocity",
+                "gyvel_right": "eye1_x_velocity",
+                "hxvel_right": "eye1_head_x_velocity",
+                "hyvel_right": "eye1_head_y_velocity",
+                "rxvel_right": "eye1_raw_x_velocity",
+                "ryvel_right": "eye1_raw_y_velocity",
+                "fgxvel": "eye1_fast_gaze_x_velocity",
+                "fgyvel": "eye1_fast_gaze_y_velocity",
+                "fhxyvel": "eye1_fast_head_x_velocity",
+                "fhyvel": "eye1_fast_head_y_velocity",
+                "frxyvel": "eye1_fast_raw_x_velocity",
+                "fryvel": "eye1_fast_raw_y_velocity",
+            }
+        )
+
+        self.samples = self.samples.replace({np.nan: "n/a"})
+        self.samples = self.samples.replace({100000000: "n/a"})
+
+        if self.task_name in ["rest", "bht", "qct"]:
+            output_file_name = f"sub-{self.participant:03d}_ses-{self.session:03d}_task-{self.task_name}_dir-{self.pe}_eyetrack.tsv.gz"
+            output_file_dir = os.path.join(
+                BIDS_folder_path,
+                f"sub-{self.participant:03d}/ses-{self.session:03d}/func/",
+            )
+        elif self.task_name == "dwi":
+            output_file_name = f"sub-{self.participant:03d}_ses-{self.session:03d}_acq-highres_dir-{self.pe}_eyetrack.tsv.gz"
+            output_file_dir = os.path.join(
+                BIDS_folder_path,
+                f"sub-{self.participant:03d}/ses-{self.session:03d}/dwi/",
+            )
+        else:
+            output_file_name = f"sub-{self.participant:03d}_ses-{self.session:03d}_task-{self.task_name}_eyetrack.tsv.gz"
+            output_file_dir = os.path.join(
+                BIDS_folder_path,
+                f"sub-{self.participant:03d}/ses-{self.session:03d}/func/",
+            )
+
+        os.makedirs(output_file_dir, exist_ok=True)
+
+        output_file_full_path = os.path.join(output_file_dir, output_file_name)
+
+        column_order = [
+            "eye_timestamp",
+            "eye1_x_coordinate",
+            "eye1_y_coordinate",
+        ] + [
+            col
+            for col in self.samples.columns
+            if col not in ["eye_timestamp", "eye1_x_coordinate", "eye1_y_coordinate"]
+        ]
+
+        self.samples = self.samples[column_order]
+        self.samples.to_csv(
+            output_file_full_path,
+            sep="\t",
+            index=False,
+            header=False,
+            compression="gzip",
+        )
+
+        return self.samples.columns.tolist()
```

Comment:
Return the filename rather than the list of columns:

```suggestion
        return output_file_full_path
```